src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java

```diff
@@ -42,55 +50,70 @@
+  @Nullable
+  @Macro
+  private String datetimeBefore;
 
   @Name(SalesforceSourceConstants.PROPERTY_DURATION)
   @Description("Salesforce SObject query duration. Time unit: hours. Default value: 0")
   @Nullable
   @Macro
-  private Integer duration;
+  private String duration;
 
   @Name(SalesforceSourceConstants.PROPERTY_OFFSET)
   @Description("Salesforce SObject query offset. Time unit: hours. Default value: 0")
   @Nullable
   @Macro
-  private Integer offset;
+  private String offset;
 
   protected SalesforceBaseSourceConfig(String referenceName,
                                        String consumerKey,
                                        String consumerSecret,
                                        String username,
                                        String password,
                                        String loginUrl,
-                                       @Nullable String datetimeFilter,
-                                       @Nullable Integer duration,
-                                       @Nullable Integer offset) {
+                                       @Nullable String datetimeAfter,
+                                       @Nullable String datetimeBefore,
+                                       @Nullable String duration,
+                                       @Nullable String offset) {
     super(referenceName, consumerKey, consumerSecret, username, password, loginUrl);
-    this.datetimeFilter = datetimeFilter;
+    this.datetimeAfter = datetimeAfter;
+    this.datetimeBefore = datetimeBefore;
     this.duration = duration;
     this.offset = offset;
   }
 
-  public int getDuration() {
-    return Objects.isNull(duration) ? SalesforceSourceConstants.DURATION_DEFAULT : duration;
+  public Map<ChronoUnit, Integer> getDuration() {
+    return Strings.isNullOrEmpty(duration) ? Collections.emptyMap() : extractRangeValue(duration);
   }
 
-  public int getOffset() {
-    return Objects.isNull(offset) ? SalesforceSourceConstants.OFFSET_DEFAULT : offset;
+  public Map<ChronoUnit, Integer> getOffset() {
+    return Strings.isNullOrEmpty(offset) ? Collections.emptyMap() : extractRangeValue(offset);
   }
 
   @Nullable
-  public String getDatetimeFilter() {
-    return datetimeFilter;
+  public String getDatetimeAfter() {
+    return datetimeAfter;
+  }
+
+  @Nullable
+  public String getDatetimeBefore() {
+    return datetimeBefore;
   }
 
   protected void validateFilters() {
-    validateSObjectFilter(SalesforceSourceConstants.PROPERTY_DURATION, getDuration());
-    validateSObjectFilter(SalesforceSourceConstants.PROPERTY_OFFSET, getOffset());
+    validateSObjectDatetimeFilter(SalesforceSourceConstants.PROPERTY_DATETIME_AFTER, getDatetimeAfter());
+    validateSObjectDatetimeFilter(SalesforceSourceConstants.PROPERTY_DATETIME_BEFORE, getDatetimeBefore());
+    validateSObjectRangeFilter(SalesforceSourceConstants.PROPERTY_DURATION, getDuration());
```

Comment:
if an exception is thrown (for example, an invalid chrono unit is given, or the integer could not be parsed), this should throw an InvalidConfigPropertyException. Same with the other fields